claims_common.go

```diff
@@ -0,0 +1,339 @@
+	// nolint
+	PsaProfile1 = "PSA_IOT_PROFILE_1"
+
+	// PsaProfile2 is the new profile in
+	// draft-tschofenig-rats-psa-token-08 and newer
+	// which uses EAT claims where possible
+	// nolint
+	PsaProfile2 = "http://arm.com/psa/2.0.0"
+)
+
+const (
+	SecurityLifecycleUnknownMin                = 0x0000
+	SecurityLifecycleUnknownMax                = 0x00ff
+	SecurityLifecycleAssemblyAndTestMin        = 0x1000
+	SecurityLifecycleAssemblyAndTestMax        = 0x10ff
+	SecurityLifecyclePsaRotProvisioningMin     = 0x2000
+	SecurityLifecyclePsaRotProvisioningMax     = 0x20ff
+	SecurityLifecycleSecuredMin                = 0x3000
+	SecurityLifecycleSecuredMax                = 0x30ff
+	SecurityLifecycleNonPsaRotDebugMin         = 0x4000
+	SecurityLifecycleNonPsaRotDebugMax         = 0x40ff
+	SecurityLifecycleRecoverablePsaRotDebugMin = 0x5000
+	SecurityLifecycleRecoverablePsaRotDebugMax = 0x50ff
+	SecurityLifecycleDecommissionedMin         = 0x6000
+	SecurityLifecycleDecommissionedMax         = 0x60ff
+)
+
+func securityLifeCycleToString(v uint16) string {
+	if v >= SecurityLifecycleUnknownMin &&
+		v <= SecurityLifecycleUnknownMax {
+		return "unknown"
+	}
+
+	if v >= SecurityLifecycleAssemblyAndTestMin &&
+		v <= SecurityLifecycleAssemblyAndTestMax {
+		return "assembly-and-test"
+	}
+
+	if v >= SecurityLifecyclePsaRotProvisioningMin &&
+		v <= SecurityLifecyclePsaRotProvisioningMax {
+		return "psa-rot-provisioning"
+	}
+
+	if v >= SecurityLifecycleSecuredMin &&
+		v <= SecurityLifecycleSecuredMax {
+		return "secured"
+	}
+
+	if v >= SecurityLifecycleNonPsaRotDebugMin &&
+		v <= SecurityLifecycleNonPsaRotDebugMax {
+		return "non-psa-rot-debug"
+	}
+
+	if v >= SecurityLifecycleRecoverablePsaRotDebugMin &&
+		v <= SecurityLifecycleRecoverablePsaRotDebugMax {
+		return "recoverable-psa-rot-debug"
+	}
+
+	if v >= SecurityLifecycleDecommissionedMin &&
+		v <= SecurityLifecycleDecommissionedMax {
+		return "decommissioned"
+	}
+
+	return "invalid"
+}
+
+func isValidSecurityLifeCycle(v uint16) error {
+	// Accept any security lifecycle in the state machine, including values that
+	// can't produce trustable PSA evidence.
+	if securityLifeCycleToString(v) == "invalid" {
+		return fmt.Errorf("%w: value %d is invalid", ErrWrongSyntax, v)
+	}
+
+	return nil
+}
+
+func isValidImplID(v []byte) error {
+	l := len(v)
+
+	if l != ImplIDLen {
+		return fmt.Errorf(
+			"%w: invalid length %d (MUST be %d bytes)",
+			ErrWrongSyntax, l, ImplIDLen,
+		)
+	}
+
+	return nil
+}
+
+func isValidBootSeed(v []byte, profile string) error {
+	l := len(v)
+
+	switch profile {
+	case PsaProfile2:
+		if l < 8 || l > 32 {
+			return fmt.Errorf(
+				"%w: invalid length %d (MUST be between 8 and 32 bytes)",
+				ErrWrongSyntax, l,
+			)
+		}
+	case PsaProfile1:
+		if l != 32 {
+			return fmt.Errorf(
+				"%w: invalid length %d (MUST be 32 bytes)",
+				ErrWrongSyntax, l,
+			)
+		}
+	}
+
+	return nil
+}
+
+func isValidCertificationReference(v string) error {
```

Comment:
I recollect this will change soon to 18 right ?

Reply:
the sentence is slightly cryptic, can you unpack it a bit?

Reply:
Ah, I mean the length of Cert reference will be 13+5 =18 in near future!

Reply:
yes, see https://github.com/thomas-fossati/draft-psa-token/issues/58